File: DMOPC2014C3P4.cpp

#include <bits/stdc++.h>
using namespace std;

int factors[100001];

int main() {
    cin.sync_with_stdio(0);
    cin.tie(0);
    for (int j = 1;j<=100000;j++) {
        
        for (int k = 1;k<=sqrt(j);k++) {
            
            if (j % k == 0 && k*k != j) {factors[j] += 2;}
            else if (k*k == j) {factors[j]++;}
        }
    }
    
    int i; scanf("%d", &i);

    for (i; i > 0;i--) {
        int K, A, B, ans = 0;
        scanf("%d%d%d", &K, &A, &B);

        
        for (int j = A;j<=B;j++) {
            if ((factors[j]) == K) ans++;
        }
        printf("%d\n", ans);
    }
}
        

File: DMOPC2014C3P4.java




import java.util.*;
import java.io.*;

public class DMOPC2014C3P4 {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringTokenizer st;

    
    public static void main(String[] args) throws NumberFormatException, IOException {
        int[] factors = new int[100001];
        for (int j = 1;j<=100000;j++) {
            
            for (int k = 1;k<=(int)Math.sqrt(j);k++) {
                
                if (j % k == 0 && k*k != j) {factors[j] += 2;}
                else if (k*k == j) {factors[j]++;}
            }
        }
        
        for (int i = Integer.parseInt(next()); i > 0;i--) {
            int K = Integer.parseInt(next()), A = Integer.parseInt(next()), B = Integer.parseInt(next()), ans = 0;
            
            for (int j = A;j<=B;j++) {
                if ((factors[j]) == K) ans++;
            }
            System.out.println(ans);
        }
           
        
        
        
    }
 
    
        
    
    static String next () throws IOException {
        while (st == null || ! st.hasMoreTokens())
            st = new StringTokenizer(br.readLine().trim());
        return st.nextToken();
    }
    
}


